tools/rostopic/src/rostopic/__init__.py

```diff
@@ -657,11 +660,19 @@ def __init__(self, topic, msg_eval, plot=False, filter_fn=None,
         self.last_topic = None
         self.last_msg_eval = None
 
-    def custom_strify_message(self, val, indent='', time_offset=None, current_time=None, field_filter=None, type_information=None, fixed_numeric_width=None):
+    def custom_strify_message(self, val, indent='', time_offset=None, current_time=None, field_filter=None, value_transform=None,
+                              type_information=None, fixed_numeric_width=None):
         # ensure to print uint8[] as array of numbers instead of string
         if type_information and type_information.startswith('uint8['):
             val = [ord(x) for x in val]
-        return genpy.message.strify_message(val, indent=indent, time_offset=time_offset, current_time=current_time, field_filter=field_filter, fixed_numeric_width=fixed_numeric_width)
+        if value_transform:
+            val, value_untransform_fn = value_transform(val)
+        text = genpy.message.strify_message(val, indent=indent, time_offset=time_offset, current_time=current_time, field_filter=field_filter, fixed_numeric_width=fixed_numeric_width)
+        if value_transform:
+            # this operation is necessary because value_transform does change the type of message
+            # and that causes not strified values next time
+            val = value_untransform_fn(val)
```

Comment:
This might be easier to read like this:

```diff
-        return genpy.message.strify_message(val, indent=indent, time_offset=time_offset, current_time=current_time, field_filter=field_filter, fixed_numeric_width=fixed_numeric_width)
 +        value_untransform_fn = None
 +        if value_transform:
 +            val, value_untransform_fn = value_transform(val)
 +        text = genpy.message.strify_message(val, indent=indent, time_offset=time_offset, current_time=current_time, field_filter=field_filter, fixed_numeric_width=fixed_numeric_width)
 +        if value_untransform_fn is not None:
 +            # this operation is necessary because value_transform does change the type of message
 +            # and that causes not strified values next time
 +            val = value_untransform_fn(val)
```